src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -441,6 +449,9 @@ public IndexState(
       initSaveLoadState();
     }
     searchThreadPoolExecutor = globalState.getSearchThreadPoolExecutor();
+
+    // add meta data fields
+    metaFields = getPredefinedMetaFields();
```

Comment:
Can this initialization be done statically. I think the meta field defs should be the same for all indices.

Reply:
nope. It will throw NPE if I make it static. SimilarityCreator.getInstance() and FieldDefCreator.getInstance() are both null otherwise. It seems we have a lot of static instance. But they are all initialized in constructors... Don't want to change all of these code here.